portsyncd/linksync.cpp

```diff
@@ -226,27 +234,24 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
     /* Insert or update the ifindex to key map */
     m_ifindexNameMap[ifindex] = key;
 
+    if (nlmsg_type == RTM_DELLINK)
+    {
+        m_statePortTable.del(key);
+        SWSS_LOG_NOTICE("Delete %s(ok) from state db", key.c_str());
+        return;
+    }
+
     /* front panel interfaces: Check if the port is in the PORT_TABLE
      * non-front panel interfaces such as eth0, lo which are not in the
      * PORT_TABLE are ignored. */
     vector<FieldValueTuple> temp;
     if (m_portTable.get(key, temp))
     {
-        /* TODO: When port is removed from the kernel */
-        if (nlmsg_type == RTM_DELLINK)
-        {
-            return;
-        }
-
-        /* Host interface is created */
-        if (!g_init && g_portSet.find(key) != g_portSet.end())
```

Comment:
>if (!g_init && g_portSet.find(key) != g_portSet.end()) [](start = 8, length = 54)

I think this check is for restart swss case. Did you tested this case?

Reply:
The steps are:
1. everything is running
2. restart swss
3. during create_switch, SAI will remove all the host interfaces and recreate them
4. the remove messages will reach here, and you need to ignore these messages

@prsunny may clarify if anything wrong.

---
In reply to: [359622619](https://github.com/Azure/sonic-swss/pull/1112#discussion_r359622619) [](ancestors = 359622619)

Reply:
@qiluo-msft @prsunny  The remove messages was done above, will not reach here
```
    if (nlmsg_type == RTM_DELLINK)
    {
        m_statePortTable.del(key);
        SWSS_LOG_NOTICE("Delete %s(ok) from state db", key.c_str());
        return;
    }
```

I think for that case, we should be fine.

Reply:
@zhenggen-xu , as Qi mentioned, this is for the restart swss case. There would be a period where the previous interfaces are present in kernel before SAI creates the new interfaces and you may get some netlink notifications during swss start. Since the check is removed, you would be adding it to state_db immediately. This is my thought!

Reply:
Based on the service file, if we restart swss service,  we stop swss docker, stop syncd docker.  Then start swss docker, start syncd docker.  It is true that swss daemons are running when syncd was starting, however,  syncd should be deleting the interfaces then add them, right?  How is it going to produce RTM_NEWLINK if it is not really creating the interface which will trigger this code to add to state_db?

Reply:
Also, we do have the code above to filter the old interfaces messages if any:
```
    if (m_ifindexOldNameMap.find(ifindex) != m_ifindexOldNameMap.end())
    {
        SWSS_LOG_INFO("nlmsg type:%d Ignoring message for old interface %s(%d)",
                nlmsg_type, key.c_str(), ifindex);
        return;
    }
```